Allow PDep reactions and non-template library reactions in PDep networks

alongd opened the pull request:
This PR allows PDep networks to use kinetics from a LibraryReaction in two cases:
1. When a unimolecular library reaction does not correspond to any kinetics family (but is flagged as `elementary`)
2. When a unimolecular library reaction has PDep kinetics, in which case a high pressure limit rate is generated and saved as a new attribute of the Reaction. This rate is then used in the network.

A test was added for the high pressure limit kinetic generation. This PR involved relatively modest changed to pdep.py, so direct pdep tests were currently not added.

alongd commented:
Addresses #1313

alongd commented:
Tests overall look good on this branch:
We have the usual non deterministic GAV interchange of [`ring(Cyclopropene)` and `ring(oxirene)`] and [`ring(cyclopropanone)` and `ring(2(co)oxirane)`].
The MCH test generated identical models, but then says `Observable species CC1CCCCC1 varied by more than 0.050 on average between old model MCH(1) and new model MCH(1) in condition 1.` The reason is unclear to me.

[main_log.txt](https://github.com/ReactionMechanismGenerator/RMG-Py/files/1987827/main_log.txt)

mjohnson541 commented on rmgpy/data/kinetics/libraryTest.py:
```diff
@@ -81,4 +83,31 @@ def testSaveLibrary(self):
+        libRxns = self.libraries['lib_net'].getLibraryReactions()
+        for rxn in libRxns:
+            self.assertIsNone(rxn.network_kinetics)
+            logging.debug(rxn)
```
If this debug statement was intended to be included I think you should make the message more specific

mjohnson541 commented on rmgpy/data/kinetics/libraryTest.py:
```diff
@@ -81,4 +83,31 @@ def testSaveLibrary(self):
+                    self.assertIsNone(rxn.network_kinetics)
+                else:
+                    self.assertTrue(isinstance(rxn.network_kinetics, Arrhenius))
+            if isinstance(rxn.kinetics,Troe):
```
Should this be under the else immediately before it?

mjohnson541 commented on rmgpy/data/kinetics/libraryTest.py:
```diff
@@ -81,4 +83,31 @@ def testSaveLibrary(self):
+            if isinstance(rxn.kinetics,Troe):
+                # some simple quantitative checks for network_kinetics
+                self.assertAlmostEqual(rxn.network_kinetics.n.value_si, 0)
+                self.assertAlmostEqual(int(rxn.network_kinetics.A.value_si), 6e+8)
```
Can you explain these checks with comments or at least in the commit?

mjohnson541 commented on rmgpy/test_data/testing_database/kinetics/libraries/lib_net/reactions.py:
```diff
@@ -0,0 +1,122 @@
+        n = 5.135,
+        Ea = (33.0118, 'kJ/mol'),
+        T0 = (1, 'K')),
+    elementary_high_p = True,
```
Why isn't it necessary to label most of the other reactions this way?

alongd replied:
It is, you are right. I wasn't consistent in defining `generate_high_p_limit_kinetics()`.
I added a check for `elementary_high_p` now.

mjohnson541 commented on rmgpy/rmg/model.py:
```diff
@@ -1429,7 +1429,16 @@ def addSeedMechanismToCore(self, seedMechanism, react=False):
                  )
             r, isNew = self.makeNewReaction(rxn) # updates self.newSpeciesList and self.newReactionlist
-            if not isNew: logging.info("This library reaction was not new: {0}".format(rxn))
+            if not isNew:
```
Why do we need this processing changes for seeds and reaction libraries?

alongd replied:
To convert each new reaction from libraries/seeds which has an Arrhenius-type kinetics (and is flagged as high-P elementary) into a PDep-type kinetics
(`processNewReactions` calls `rxn.generate_high_p_limit_kinetics()`)

mjohnson541 commented on documentation/source/users/rmg/database/modification.rst:
```diff
@@ -81,6 +81,25 @@ To conform to RMG's format, simply copy and modify an existing library.
+
+2. **has_pdep_route**: If set to ``True`` and RMG discovers a pressure-dependent reaction with the same reactants and products,
+the latter *will* be considered in addition to the library reaction. This is useful for cases when more than one pathway connects
+the samethe same reactants and products, and some of these pathways are well-skipping reactions. If set to ``False``,
```
typo

mjohnson541 commented on documentation/source/users/rmg/database/modification.rst:
```diff
@@ -81,6 +81,25 @@ To conform to RMG's format, simply copy and modify an existing library.
+2. **has_pdep_route**: If set to ``True`` and RMG discovers a pressure-dependent reaction with the same reactants and products,
+the latter *will* be considered in addition to the library reaction. This is useful for cases when more than one pathway connects
+the samethe same reactants and products, and some of these pathways are well-skipping reactions. If set to ``False``,
+similar network reactions will not be considered in the model generation.
```
What is the default value?

mjohnson541 commented on documentation/source/users/rmg/database/modification.rst:
```diff
@@ -81,6 +81,25 @@ To conform to RMG's format, simply copy and modify an existing library.
+limit rate in network exploration, and will not convert Arrhenius kinetics of library reactions that have no template
+(a corresponding reaction family) into a pressure-dependent form.
+
+2. **has_pdep_route**: If set to ``True`` and RMG discovers a pressure-dependent reaction with the same reactants and products,
```
Would "allow_pdep_route" be a better name?

alongd commented:
Thanks, @mjohnson541 
I addressed your comments in fixup commits. The last commit isn't a fixup, since `allow_pdep_route` isn't part of this PR (I just added the documentation for it here).
Let me know when to squash

mjohnson541 commented:
Ok this looks good...even Codacy is happy!

alongd commented:
Sweet! OK, rebased and squashed, ready to merge once tests pass (please re-approve)

mliu49 commented:
I think you'll need to rebase and reapprove again since you just merged the other PR...

mjohnson541 commented:
Yes, but that doesn't seem to have prevented me from approving it again...

alongd commented:
Please re-approve... :)
